test/cppsuite/test_harness/workload/database_operation.cxx

```diff
@@ -49,16 +49,14 @@ populate_worker(thread_context *tc)
          * is closed, WiredTiger APIs close the cursors too.
          */
         scoped_cursor cursor = tc->session.open_scoped_cursor(coll.name.c_str());
-        for (uint64_t i = 0; i < tc->key_count; ++i) {
-            /* Start a txn. */
+        uint64_t j = 0;
+        while (j < tc->key_count) {
             tc->transaction.begin();
-            if (tc->insert(cursor, coll.id, i)) {
-                /* We failed to insert, rollback our transaction and retry. */
-                tc->transaction.rollback();
-                --i;
-                continue;
+            if (tc->insert(cursor, coll.id, j)) {
+                if (tc->transaction.commit()) {
```

Comment:
Could we do the following logic instead of the current one:

- `break` if `insert` fails
- `commit` outside the while loop if `j >= tc->key_count`
- `testutil_assert(commit == true)`

Reply:
Whats the goal of that logic, there's no guarantee your commit will succeed either?

Reply:
With the current implementation, we could stay blocked in the while loop.

Reply:
I've ran into cases where the commit fails due to cache pressure, I don't think we want to die there if we have some cache pressure. So can we leave it as it for now?

Reply:
Ok, we are coming back to the idea of having a variable that sets a maximum number of times we can fail. Let's leave it for now.